src/Tests/Microsoft.DotNet.PackageValidation.Tests/ValidatePackageTargetTests.cs

```diff
@@ -0,0 +1,97 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.IO;
+using Microsoft.NET.TestFramework;
+using Microsoft.NET.TestFramework.Commands;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.DotNet.PackageValidation.Tests
+{
+    public class ValidatePackageTargetTests : SdkTest
+    {
+        public ValidatePackageTargetTests(ITestOutputHelper log) : base(log)
+        {
+            // clearing the cache
+            string localSdkNugetCacheValidationPath = Path.Combine(TestContext.Current.NuGetCachePath, "microsoft.dotnet.packagevalidation");
+            if (Directory.Exists(localSdkNugetCacheValidationPath))
+                Directory.Delete(localSdkNugetCacheValidationPath, recursive: true);
+
+            // Packing and copying the local version package validation
+            string packageValidationProjectPath = Path.Combine(Path.GetDirectoryName(_testAssetsManager.TestAssetsRoot), "Compatibility", "Microsoft.DotNet.PackageValidation", "Microsoft.DotNet.PackageValidation.csproj");
+            new PackCommand(Log, packageValidationProjectPath)
+                .Execute($"-p:PackageVersion=1.0.0-test;PackageOutputPath={TestContext.Current.TestPackages}");
+        }
+
+        [Fact]
+        public void ValidatePackageTargetRunsSuccessfully()
+        {
+            var testAsset = _testAssetsManager
+                .CopyTestAsset("PackageValidationTestProject")
+                .WithSource();
+
+            var result = new PackCommand(Log, Path.Combine(testAsset.TestRoot, "PackageValidationTestProject.csproj"))
+                .Execute();
+
+            Assert.Equal(0, result.ExitCode);
+            Assert.Contains(Resources.SuccessfulPackageRun, result.StdOut);
```

Comment:
Perhaps instead of checking for a string in the output to validate if we actually ran, we should have the validation target to output a  semaphore file on the obj directory which could also be used as inputs/outputs of the target so that we can skip it if the package didn't really change. Then we could just check here in the test if the semaphore was written.